ui/src/app/views/homepage/component.tsx

```diff
@@ -0,0 +1,394 @@
+import {ClrIcon} from 'app/components/icons';
+import {profileApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, withStyle} from 'app/utils';
+import {navigateByUrl} from 'app/utils/navigation';
+import {
+  BillingProjectStatus,
+  IdVerificationStatus,
+  PageVisit,
+  Profile,
+  ProfileService
+} from 'generated';
+
+
+const styles = reactStyles({
+  mainHeader: {
+    color: '#FFFFFF',
+    fontSize: 28,
+    fontWeight: 400,
+    width: '25.86%',
+    display: 'flex',
+    minWidth: '18.2rem',
+    marginLeft: '4%',
+    marginTop: '4%',
+    letterSpacing: 'normal'
+  },
+  minorHeader: {
+    color: '#FFFFFF',
+    fontSize: 18,
+    fontWeight: 600,
+    display: 'flex',
+    marginTop: '1rem',
+    lineHeight: '24px'
+  },
+  text: {
+    color: '#FFFFFF',
+    fontSize: 16,
+    lineHeight: '22px',
+    fontWeight: 150,
+    marginTop: '3%'
+  },
+  infoBox: {
+    padding: '1rem',
+    backgroundColor: '#FFFFFF',
+    borderRadius: '5px',
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between'
+  },
+  infoBoxHeader: {
+    fontSize: 16,
+    color: '#262262',
+    fontWeight: 600
+  },
+  infoBoxBody: {
+    color: '#000',
+    lineHeight: '18px'
+  },
+  infoBoxButton: {
+    color: '#FFFFFF',
+    height: '49px',
+    borderRadius: '5px',
+    marginLeft: '1rem',
+    maxWidth: '20rem'
+  },
+  error: {
+    fontWeight: 300,
+    color: '#DC5030',
+    fontSize: '14px',
+    marginTop: '.3rem',
+    backgroundColor: '#FFF1F0',
+    borderRadius: '5px',
+    padding: '.3rem',
+  }
+});
+
+export const Error = withStyle(styles.error)('div');
+
+const AccountLinkingButton: React.FunctionComponent<{
+  failed: boolean, completed: boolean, failedText: string,
+  completedText: string, defaultText: string, onClick: Function
+}> = ({failed, completed, defaultText, completedText, failedText, onClick}) => {
+  if (failed) {
+    return <Clickable style={{...buttonStyles.base,
+      ...styles.infoBoxButton, backgroundColor: '#f27376'}} disabled={true}>
+      <ClrIcon shape='exclamation-triangle'/>{failedText}
+    </Clickable>;
+  } else if (completed) {
+    return <Clickable style={{...buttonStyles.base,
+      ...styles.infoBoxButton, backgroundColor: '#8BC990'}} disabled={true}>
+      <ClrIcon shape='check'/>{completedText}
+    </Clickable>;
+  } else {
+    return <Clickable style={{...buttonStyles.base,
+      ...styles.infoBoxButton, backgroundColor: '#2691D0'}}
+                      onClick={onClick}>
+      {defaultText}
+    </Clickable>;
+  }
+};
+
+export interface AccountLinkingProps {
+  eraCommonsLinked: boolean;
+  trainingCompleted: boolean;
+}
+
+export class AccountLinking extends
+    React.Component<AccountLinkingProps, {eraCommonsLinked: boolean, eraCommonsError: string}> {
+
+  constructor(props: AccountLinkingProps) {
+    super(props);
+
+    this.state = {
+      eraCommonsError: '',
+      eraCommonsLinked: props.eraCommonsLinked
```

Comment:
Why put this into state? It never changes.

Reply:
When we return from the redirect, that's a hard page reload so this client state is no longer relevant. I don't see any `setState` affecting `eraCommonsLinked` in this file.

Reply:
this has been resolved